nautobot/extras/tests/test_relationships.py

```diff
@@ -384,6 +384,38 @@ def setUp(self):
         for cra in self.invalid_relationship_associations:
             cra.validated_save()
 
+    def test_create_invalid_relationship_association(self):
+        # Test creation of invalid relationship association restricted by destination/source filter
+
+        relationship = Relationship.objects.create(
+            name="Site to Rack Rel 1",
+            slug="site-to-rack-rel-1",
+            source_type=self.site_ct,
+            source_filter={"name": [self.sites[0].name]},
+            destination_type=self.rack_ct,
+            destination_label="Primary Rack",
+            type=RelationshipTypeChoices.TYPE_ONE_TO_ONE,
+            destination_filter={"name": [self.racks[0].name]},
+        )
+
+        associations = (
+            (
+                "source",
+                RelationshipAssociation(relationship=relationship, source=self.sites[1], destination=self.racks[0]),
+            ),
+            (
+                "destination",
+                RelationshipAssociation(relationship=relationship, source=self.sites[0], destination=self.racks[1]),
+            ),
+        )
+
+        for side_name, association in associations:
+            side = getattr(association, side_name)
+            with self.assertRaises(ValidationError) as handler:
+                association.validated_save()
+            expected_errors = {side_name: [f"{side} violates {relationship} {side_name}_filter restriction"]}
+            self.assertEqual(handler.exception.message_dict, expected_errors)
+
```

Comment:
Given the original issue described an issue with the REST API, does it make sense to add an API-specific test case, even if it's somewhat redundant with this ORM test case?

Reply:
I don't think it's needed